### PR TITLE
[mssql] use nullable info from db to set field ConstraintNotNull

### PR DESCRIPTION
**Pull request by domi4484**
Set ConstraintNotNull also for other fields than primary key according to informations from sql server

### CODE REVIEW

**Comment by m-kuhn** [comment]
Can you confirm you did run the tests locally (I guess they are not run in CI)

File: src/providers/mssql/qgsmssqlprovider.cpp [hunk 1]
```diff
@@ -364,175 +364,186 @@ void QgsMssqlProvider::loadFields()
...
-            QgsField(
-              colName, sqlType,
-              sqlTypeName ) );
-        }
+      // Field nullable
+      const bool nullable = query.value( 10 ).toBool();
 
-        //COLUMN_DEF
-        if ( !query.value( 12 ).isNull() )
+      QgsDebugMsg( QString( "Col name: %1, Nullable: %2" ).arg( colName ).arg( query.value( 10 ).toString() ) );
```
Comment (m-kuhn) [inline src/providers/mssql/qgsmssqlprovider.cpp:411]:
```suggestion
      QgsDebugMsg( QStringLiteral( "Col name: %1, Nullable: %2" ).arg( colName ).arg( query.value( 10 ).toString() ) );
```

File: src/providers/mssql/qgsmssqlprovider.cpp [hunk 2]
```diff
@@ -364,175 +364,186 @@ void QgsMssqlProvider::loadFields()
...
-              -1 ) );
-        }
-        else
-        {
-          mAttributeFields.append(
-            QgsField(
-              colName, sqlType,
-              sqlTypeName ) );
-        }
+      // Field nullable
```
Comment (m-kuhn) [inline src/providers/mssql/qgsmssqlprovider.cpp:408]:
```suggestion
      // 10 means: field nullable
```

is that correct?
hooray magic numbers ... :stuck_out_tongue_closed_eyes:

File: src/providers/mssql/qgsmssqlprovider.cpp [hunk 3]
```diff
@@ -364,175 +364,186 @@ void QgsMssqlProvider::loadFields()
...
-          mDefaultValues.insert( i, query.value( 12 ).toString() );
+          length = length / 2;
         }
-
-        ++i;
+        QgsField field = QgsField(
+                           colName, sqlType,
+                           sqlTypeName,
+                           length );
+        if ( !nullable )
```
Comment (m-kuhn) [inline src/providers/mssql/qgsmssqlprovider.cpp:430]: Is this check needed? As soon as we add more constraints like unique I guess that's obsolete

**Comment by domi4484** [comment]
@m-kuhn Yes I ran the test locally.
Some unrelated tests where failing already before my changes. Did someone now if all the mssql provider test are broken?
- testExtentFromGeometryTable
- testPktComposite
- testPktCompositeFloat
These are failing, but maybe i am just missing something in my setup?

File: src/providers/mssql/qgsmssqlprovider.cpp (hunk 3 above)
Comment (domi4484) [reply src/providers/mssql/qgsmssqlprovider.cpp:430]: yes I can remove it

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**